docs/data/material/components/text-fields/InputAdornments.js

```diff
@@ -13,24 +13,9 @@ import Visibility from '@mui/icons-material/Visibility';
 import VisibilityOff from '@mui/icons-material/VisibilityOff';
 
 export default function InputAdornments() {
-  const [values, setValues] = React.useState({
-    amount: '',
-    password: '',
-    weight: '',
-    weightRange: '',
-    showPassword: false,
-  });
+  const [showPassword, setShowPassword] = React.useState(false);
 
-  const handleChange = (prop) => (event) => {
-    setValues({ ...values, [prop]: event.target.value });
-  };
-
-  const handleClickShowPassword = () => {
-    setValues({
-      ...values,
-      showPassword: !values.showPassword,
-    });
-  };
+  const handleClickShowPassword = () => setShowPassword(!showPassword);
```

Comment:
Changing state based on a previous value should use the callback form:
```suggestion
  const handleClickShowPassword = () => setShowPassword((show) => !show);
```